machine/vm.h

```diff
@@ -11,25 +11,27 @@
 # define SATP_MODE_CHOICE INSERT_FIELD(0, SATP64_MODE, SATP_MODE_SV39)
 # define VA_BITS 39
 # define GIGAPAGE_SIZE (MEGAPAGE_SIZE << RISCV_PGLEVEL_BITS)
+typedef uint64_t addr_t;
 #else
 # define SATP_MODE_CHOICE INSERT_FIELD(0, SATP32_MODE, SATP_MODE_SV32)
 # define VA_BITS 32
+typedef uint32_t addr_t;
```

Comment:
This should probably be `vaddr_t`, since SV32 maps 32-bit virtual addresses to 34-bit physical addresses.